erpnext/subcontracting/doctype/subcontracting_order/subcontracting_order.js

```diff
@@ -138,6 +138,17 @@ frappe.ui.form.on('Subcontracting Order', {
 	}
 });
 
+frappe.ui.form.on('Subcontracting Order Supplied Item', {
+	sourced_by_supplier: function(frm, cdt, cdn) {
+		let row = locals[cdt][cdn];
+		row.rate = row.amount = 0;
+		frm.refresh_field("supplied_items");
+		frm.call('update_raw_material_cost').then(r => {
+			frm.refresh_field("items");
+		});
```

Comment:
Do we need to go to the backend for this? If we change this directly in JS, would save handle it?

Reply:
It can be done in JS and save will also handle it, I'll remove `update_raw_material_cost` because there's a function `set_missing_values_in_supplied_items` which will handle it just need to update it by adding check for sourced_by_supplier.